src/toil/wdl/wdltoil.py

```diff
@@ -1578,6 +1587,46 @@ def map_over_typed_files_in_value(value: WDL.Value.Base, transform: Callable[[WD
         # All other kinds of value can be passed through unmodified.
         return value
 
+def map_over_files_in_value_check_null_type(value: WDL.Value.Base, original_value: WDL.Value.Base, expected_type: WDL.Type.Base) -> None:
```

Comment:
We maybe should cut `map_over_files_in_value` out of the name here, since this doesn't take a function to map. Maybe we want to call it something like `ensure_null_files_are_nullable` or something.